src/prefect/results.py

```diff
@@ -247,6 +247,56 @@ async def update_for_task(self: Self, task: "Task") -> Self:
             update["result_storage"] = await get_default_result_storage()
         return self.model_copy(update=update)
 
+    @sync_compatible
+    async def _exists(self, key: str) -> bool:
+        """
+        Check if a result record exists in storage.
+
+        Args:
+            key: The key to check for the existence of a result record.
+
+        Returns:
+            bool: True if the result record exists, False otherwise.
+        """
+        if self.metadata_storage is not None:
+            # TODO: Add an `exists` method to commonly used storage blocks
+            # so the entire payload doesn't need to be read
+            try:
+                metadata_content = await self.metadata_storage.read_path(key)
```

Comment:
for the future, i think this may have to do a second lookup on the result itself unfortunately; the metadata could point to results that no longer exist, and I wouldn't want that to error out and cause execution disruption. I don't think it needs to be addressed in this PR though given that no one is separating these yet